juriscraper/pacer/http.py

```diff
@@ -0,0 +1,170 @@
+"""
+Functions for Authenticating with PACER
+"""
+import re
+import requests
+
+from juriscraper.lib.log_tools import make_default_logger
+
+logger = make_default_logger()
+
+
+class PacerSession(requests.Session):
+    """
+    Extension of requests.Session to handle PACER oddities making it easier
+    for folks to just POST data to PACER endpoints/apis
+    """
+
+    def __init__(self, pacer_token=None, cookie_jar=None):
+        """
+        Instantiate a new PACER HTTP Session with some Juriscraper defaults
+        :param pacer_token: a PACER_SESSION token value
+        """
+        super(PacerSession, self).__init__()
+        self.headers['User-Agent'] = 'Juriscraper'
+        self.verify = False
+
+        if pacer_token:
+            self.cookies.set('PacerSession',
+                             pacer_token,
+                             domain='.uscourts.gov',
+                             path='/')
+
+        if cookie_jar:
+            self.cookies = cookie_jar
+
+    def post(self, url, data=None, json=None, **kwargs):
+        """
+        Overrides requests.Session.post with PACER-specific fun.
+
+        Will automatically convert data dict into proper multi-part form data
+        and pass to the files parameter instead.
+
+        Will set a timeout of 300 if not provided.
+
+        All other uses or parameters will pass through untouched
+        :param url: url string to post to
+        :param data: post data
+        :param json: json object to post
+        :param kwargs: assorted keyword arguments
+        :return: requests.Response
+        """
+        kwargs.setdefault('timeout', 300)
+
+        if data:
+            pacer_data = self._prepare_multipart_form_data(data)
+            return super(PacerSession, self).post(url, files=pacer_data, **kwargs)
+
+        return super(PacerSession, self).post(url, data=data, json=json, **kwargs)
+
+    @staticmethod
+    def _prepare_multipart_form_data(data):
+        """
+        Transforms a data dictionary into the multi-part form data that PACER
+        expects as the POST body
+        :param data: dict of data to transform
+        :return: dict with values wrapped into tuples like:(None, <value>)
+        """
+        output = dict()
+        for key in data:
+            output[key] = (None, data[key])
```

Comment:
Should be fine though since I'm using `requests==2.9.1` which is the currently used on on master. I believe the issue was more dealing with a Py2 vs Py3 issue in the underlying `urllib3` package that's embedded in `requests` (which is odd as it's also a separate project itself).